src/deploy.js

```diff
@@ -162,7 +163,7 @@ const deploy = async (opts) => {
   );
 
   // ensure the correct distribution variables are defined
-  if (!isDefined(DISTRIBUTION)) {
+  if (_.isUndefined(DISTRIBUTION)) {
```

Comment:
everything after line 159 should be inside the `on('end', () => {` call in order to work. Preferably you can use the promises version of the native `s3` client from AWS.